pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -830,4 +830,8 @@ self: super: builtins.intersectAttrs super {
       export HOME=$TMPDIR
     '';
   });
+
+  # tests depend on a specific version of solc,
+  # original derivation expects libff to be under ff attribute, add libff explicitly
+  hevm = dontCheck (doJailbreak (addExtraLibrary super.hevm pkgs.libff));
```

Comment:
This `addExtraLibrary` shouldn't be here.

The problem here is that `cabal2nix` doesn't know about this `libff` library.

What you need to do is send a PR to `master` adding `libff`, and then a separate PR to https://github.com/NixOS/cabal2nix creating this `ff` to `libff` mapping.

Here is the function you have to add to:

https://github.com/NixOS/cabal2nix/blob/d19490223d91a2db1708e423eb205884e9a9d558/src/Distribution/Nixpkgs/Haskell/FromCabal/Name.hs#L16-L27

Once both of these are merged in, you can finally send a PR adding `dontCheck` and `doJailbreak` for `hevm`.  (Although keep in mind that if you send a PR adding `doJailbreak` for `hevm`, we'll ask you to send a PR / create an issue upstream reporting that they need to loosen their version bounds, so you might want to do that in advance.)

Reply:
@cdepillabout here is the libff PR https://github.com/NixOS/nixpkgs/pull/105400 and cabal2nix PR https://github.com/NixOS/cabal2nix/pull/471